src-python/drbench/drbench/plotter.py

```diff
@@ -2,8 +2,140 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
-# TODO: stuff related to plottimng is going to be written here
+import json
+import os
+import re
+from collections import namedtuple
 
+import click
+import pandas as pd
+import seaborn as sns
+from drbench import common
 
-def do_nothing():
-    pass
+# only common_config for now, add plotting options here if needed
+PlottingConfig = namedtuple(
+    'PlottingConfig',
+    'common_config',
+)
+
+
+class Plotter:
+    @staticmethod
+    def __is_our_file(fname: str, analysis_id: str):
+        files_prefix = common.analysis_file_prefix(analysis_id)
+        if not fname.startswith(files_prefix):
+            return False
+        if fname.startswith(files_prefix + '.rank000'):
+            return True
+        if fname.startswith(files_prefix + '.rank'):
+            return False
+        return True
+
+    @staticmethod
+    def __import_file(fname: str, rows):
+        mode = re.search(r'AnalysisMode\.([A-Z_]+)\.', fname).group(1)
+        with open(fname) as f:
+            fdata = json.load(f)
+            ctx = fdata['context']
+            vsize = int(ctx['default_vector_size'])
+            nprocs = int(ctx['ranks'])
+            benchs = fdata['benchmarks']
+            for b in benchs:
+                bname = b['name'].partition('/')[0]
+                rtime = b['real_time']
+                bw = b['bytes_per_second']
+                rows.append(
+                    {
+                        'mode': mode,
+                        'vsize': vsize,
+                        'test': bname,
+                        'nprocs': nprocs,
+                        'rtime': rtime,
+                        'bw': bw,
+                    }
+                )
+
+    # db is created which looks something like this:
+    #           mode  vsize          test  nprocs      rtime            bw
+    # 0   MHP_NOSYCL  20000   Stream_Copy       1   0.234987  1.361779e+11
+    # 1   MHP_NOSYCL  20000  Stream_Scale       1   0.240879  1.328468e+11
+    # 2   MHP_NOSYCL  20000    Stream_Add       1   0.329298  1.457645e+11
+    # ..         ...    ...           ...     ...        ...           ...
+    # 62     MHP_GPU  40000    Stream_Add       4  21.716973  4.420506e+09
+    # 63     MHP_GPU  40000  Stream_Triad       4  21.714421  4.421025e+09
+    def __init__(self, plotting_config: PlottingConfig):
+        rows = []
+        for fname in os.listdir('.'):
```

Comment:
```suggestion
        for fname in glob.glob('dr-bench*.rank'):
```
It's a unix-style file path RE. https://docs.python.org/3/library/glob.html
instead of __is_our_file
Not a big deal if you prefer code instead.

Reply:
ok for using 'glob', not ok for `.rank` inside glob because shp output does not have `rank`, so it has to be `__is_our_file` in order to not have very complex regexp